pkg/addon-operator/ensure_crds.go

```diff
@@ -0,0 +1,222 @@
+package addon_operator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimachineryv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apimachineryYaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/flant/addon-operator/pkg/module_manager/models/modules"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/kube-client/client"
+)
+
+var crdGVR = schema.GroupVersionResource{
+	Group:    "apiextensions.k8s.io",
+	Version:  "v1",
+	Resource: "customresourcedefinitions",
+}
+
+func (op *AddonOperator) EnsureCRDs(module *modules.BasicModule) error {
+	// do not ensure CRDs if there are no files
+	if len(module.GetCRDFilesPaths()) == 0 {
+		return nil
+	}
+
+	result := new(multierror.Error)
+	cp, err := NewCRDsInstaller(op.KubeClient(), module.GetCRDFilesPaths(), op.CrdExtraLabels)
+	if err != nil {
+		result = multierror.Append(result, err)
+		return result
+	}
+
+	if cp == nil {
+		return nil
+	}
+
+	return cp.Run(context.TODO()).ErrorOrNil()
+}
+
+// CRDsInstaller simultaneously installs CRDs from specified directory
+type CRDsInstaller struct {
+	k8sClient     dynamic.Interface
+	crdFilesPaths []string
+	buffer        []byte
+
+	// concurrent tasks to create resource in a k8s cluster
+	k8sTasks *multierror.Group
+
+	crdExtraLabels map[string]string
+}
+
+func (cp *CRDsInstaller) Run(ctx context.Context) *multierror.Error {
+	result := new(multierror.Error)
+
+	for _, crdFilePath := range cp.crdFilesPaths {
+		if match := strings.HasPrefix(filepath.Base(crdFilePath), "doc-"); match {
```

Comment:
I guess we can exclude these `doc-xxx` files on discovery phase

Reply:
DONE